Zend/zend.c

```diff
@@ -85,7 +85,7 @@ ZEND_API void (*zend_error_cb)(int type, zend_string *error_filename, const uint
 void (*zend_printf_to_smart_string)(smart_string *buf, const char *format, va_list ap);
 void (*zend_printf_to_smart_str)(smart_str *buf, const char *format, va_list ap);
 ZEND_API char *(*zend_getenv)(const char *name, size_t name_len);
-ZEND_API zend_string *(*zend_resolve_path)(zend_string *filename);
+ZEND_API zend_string *(*zend_resolve_path)(const zend_string *filename);
```

Comment:
In general, ``zend_resolve_path()`` may return the same string and in case of non-interned string this requires incrementation if reference counter.

Reply:
Right, this makes sense, but are we not doing this in practice because [``tsrm_realpath()``](https://heap.space/s?refs=tsrm_realpath&project=php-src) still returns a ``char *`` ?

Also, shouldn't the ZendAccelerator implementation *also* increment the refcount? Instead of doing what it does by returning the pointer directly instead of ``zend_string_copy(str);``?

Reply:
``zend_string_copy()`` just increment the refcount and return the same string. See https://github.com/php/php-src/blob/master/Zend/zend_string.h#L236

Reply:
> Right, this makes sense, but are we not doing this in practice because [`tsrm_realpath()`](https://heap.space/s?refs=tsrm_realpath&project=php-src) still returns a `char *` ?

Yes. It would be great to switch to ``zend_string`` in all related functions, but this is going to be a big and not simple patch.

Reply:
> `zend_string_copy()` just increment the refcount and return the same string. See https://github.com/php/php-src/blob/master/Zend/zend_string.h#L236

Yes I know this, my question is *why* in ZendAccelerator.c the string is returned directly without incrementing the refcount, as after your explanation it seems this should be done.